app/factory.py

```diff
@@ -43,23 +43,27 @@ def get_user(self, identifier):
             if rv is not None:
                 return rv
 
+user_datastore = DeploySQLAlchemyUserDatastore(db, User, Role)
 
-def create_app(): #pylint: disable=too-many-statements
+def create_app(config=None): #pylint: disable=too-many-statements
```

Comment:
We want to be able to have fine-grained control of the app config for testing, and we don't want testing to be dependent on `local_config.yml`, so we'll allow an optional config dict to be passed in for testing to override the default behavior.